src/app/filecoin-plus/allocators/components/SelectFilter.tsx

```diff
@@ -0,0 +1,78 @@
+'use client'
+
+import {
+  Listbox as HeadlessUIListbox,
+  ListboxButton as HeadlessUIListboxButton,
+  ListboxOptions as HeadlessUIListboxOptions,
+  ListboxOption as HeadlessUIListboxOption,
+} from '@headlessui/react'
+import { CaretDown, Check } from '@phosphor-icons/react/dist/ssr'
+import type { Column } from '@tanstack/react-table'
+
+import { Icon } from '@/components/Icon'
+
+import type { Allocator } from '../schemas/AllocatorSchema'
+
+type FilterOption = {
+  id: string
+  name: string
+}
+
+type SortListboxProps = {
+  column: Column<Allocator>
+  options: ReadonlyArray<FilterOption>
+  resetOptionName: string
+}
+
+const EMPTY_FILTER_VALUE = ''
+
+export function SelectFilter({
+  column,
+  options,
+  resetOptionName,
+}: SortListboxProps) {
+  const currentFilterValue =
+    column.getFilterValue()?.toString() || EMPTY_FILTER_VALUE
```

Comment:
Do we need `|| EMPTY_FILTER_VALUE`?

Reply:
The `toString()` part is to make TypeScript happy as the return type of `getFilterValue` is `unknown`.

https://tanstack.com/table/v8/docs/api/features/column-filtering#getfiltervalue

Reply:
But it looks like we don't need it anymore in this scenario! 👌